src/ziggurat/streams.clj

```diff
@@ -53,47 +50,27 @@
-    (if (some? key-deserializer-encoding)
+    (when (some? key-deserializer-encoding)
       (.put properties KEY_DESERIALIZER_ENCODING key-deserializer-encoding))
-    (if (some? value-deserializer-encoding)
+    (when (some? value-deserializer-encoding)
       (.put properties VALUE_DESERIALIZER_ENCODING value-deserializer-encoding))
-    (if (some? deserializer-encoding)
+    (when (some? deserializer-encoding)
       (.put properties DESERIALIZER_ENCODING deserializer-encoding))))
 
 (defn- validate-auto-offset-reset-config
   [auto-offset-reset-config]
-  (if-not (contains? #{"latest" "earliest" nil} auto-offset-reset-config)
+  (when-not (contains? #{"latest" "earliest" nil} auto-offset-reset-config)
     (throw (ex-info "Stream offset can only be latest or earliest" {:offset auto-offset-reset-config}))))
 
-(defn- properties [{:keys [application-id
-                           bootstrap-servers
-                           stream-threads-count
-                           auto-offset-reset-config
-                           buffered-records-per-partition
-                           commit-interval-ms
-                           upgrade-from
-                           changelog-topic-replication-factor
-                           default-key-serde
-                           default-value-serde
-                           key-deserializer-encoding
-                           value-deserializer-encoding
-                           deserializer-encoding
-                           session-timeout-ms-config
-                           default-api-timeout-ms-config]}]
+(defn- properties [{:keys
+                    [auto-offset-reset-config
+                     upgrade-from
+                     key-deserializer-encoding
+                     value-deserializer-encoding
+                     deserializer-encoding] :as m}]
   (validate-auto-offset-reset-config auto-offset-reset-config)
-  (doto (Properties.)
-    (.put StreamsConfig/APPLICATION_ID_CONFIG application-id)
-    (.put StreamsConfig/BOOTSTRAP_SERVERS_CONFIG bootstrap-servers)
-    (.put StreamsConfig/NUM_STREAM_THREADS_CONFIG (int stream-threads-count))
-    (.put StreamsConfig/DEFAULT_KEY_SERDE_CLASS_CONFIG default-key-serde)
-    (.put StreamsConfig/DEFAULT_VALUE_SERDE_CLASS_CONFIG default-value-serde)
+  (doto (build-properties m)
     (.put StreamsConfig/DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG IngestionTimeExtractor)
-    (.put StreamsConfig/BUFFERED_RECORDS_PER_PARTITION_CONFIG (int buffered-records-per-partition))
-    (.put StreamsConfig/COMMIT_INTERVAL_MS_CONFIG commit-interval-ms)
-    (.put StreamsConfig/REPLICATION_FACTOR_CONFIG (int changelog-topic-replication-factor))
-    (.put ConsumerConfig/AUTO_OFFSET_RESET_CONFIG auto-offset-reset-config)
-    (.put ConsumerConfig/SESSION_TIMEOUT_MS_CONFIG (int session-timeout-ms-config))
-    (.put ConsumerConfig/DEFAULT_API_TIMEOUT_MS_CONFIG (int default-api-timeout-ms-config))
     (set-upgrade-from-config upgrade-from)
```

Comment:
`(set-upgrade-from-config upgrade-from)` can be removed once you do a trim and blank check while building properties.

Reply:
removed this and the encoding function also..

Reply:
That encoding function was required to support JSON based ser-des operations. We can remove that in future commits if found irrelevant

Reply:
yeah but it just checks if the key is present and sets the property if its present